src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -732,42 +743,45 @@ bool StorageRabbitMQ::checkDependencies(const StorageID & table_id)
 
 void StorageRabbitMQ::streamingToViewsFunc()
 {
-    try
+    if (rabbit_is_ready)
     {
-        auto table_id = getStorageID();
-
-        // Check if at least one direct dependency is attached
-        size_t dependencies_count = DatabaseCatalog::instance().getDependencies(table_id).size();
-
-        if (dependencies_count)
+        try
         {
-            auto start_time = std::chrono::steady_clock::now();
+            auto table_id = getStorageID();
 
-            // Keep streaming as long as there are attached views and streaming is not cancelled
-            while (!stream_cancelled && num_created_consumers > 0)
-            {
-                if (!checkDependencies(table_id))
-                    break;
+            // Check if at least one direct dependency is attached
+            size_t dependencies_count = DatabaseCatalog::instance().getDependencies(table_id).size();
 
-                LOG_DEBUG(log, "Started streaming to {} attached views", dependencies_count);
-
-                if (streamToViews())
-                    break;
+            if (dependencies_count)
+            {
+                auto start_time = std::chrono::steady_clock::now();
 
-                auto end_time = std::chrono::steady_clock::now();
-                auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end_time - start_time);
-                if (duration.count() > MAX_THREAD_WORK_DURATION_MS)
+                // Keep streaming as long as there are attached views and streaming is not cancelled
+                while (!stream_cancelled && num_created_consumers > 0)
                 {
-                    event_handler->updateLoopState(Loop::STOP);
-                    LOG_TRACE(log, "Reschedule streaming. Thread work duration limit exceeded.");
-                    break;
+                    if (!checkDependencies(table_id))
+                        break;
+
+                    LOG_DEBUG(log, "Started streaming to {} attached views", dependencies_count);
+
+                    if (streamToViews())
+                        break;
+
+                    auto end_time = std::chrono::steady_clock::now();
+                    auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end_time - start_time);
+                    if (duration.count() > MAX_THREAD_WORK_DURATION_MS)
+                    {
+                        event_handler->updateLoopState(Loop::STOP);
+                        LOG_TRACE(log, "Reschedule streaming. Thread work duration limit exceeded.");
+                        break;
+                    }
                 }
             }
         }
-    }
-    catch (...)
-    {
-        tryLogCurrentException(__PRETTY_FUNCTION__);
+        catch (...)
+        {
+            tryLogCurrentException(__PRETTY_FUNCTION__);
+        }
     }
```

Comment:
Log something that we are not ready yet?

Reply:
No, because that flag `rabbit_is_ready == false`  also means that there is no connection, so there will be enough logs about connection unavailability and reconnecting attempts.